packages/shared/src/chunker.ts

```diff
@@ -0,0 +1,49 @@
+// No clue why, but 3600 matches 4kb in the browser
+const MAX_CHUNK_SIZE = 3600;
+
+interface Chunk {
+	name: string;
+	value: string;
+}
+
+// Get cookie as chunks to set in the browser
+export function createChunks(key: string, value: string) {
+	// check the length and then split
+	const chunkCount = Math.ceil(value.length / MAX_CHUNK_SIZE);
+
+	if (chunkCount === 1) {
+		return value;
+	}
+
+	const chunks: Chunk[] = [];
+	const values = value.match(new RegExp('.{1,' + MAX_CHUNK_SIZE + '}', 'g'));
+
+	values?.forEach((value, i) => {
+		const name: string = `${key}.${i}`;
+		chunks.push({ name, value });
+	});
```

Comment:
You can also write this without regexp, but I don't mind which one you use.
```suggestion
	let cookie = Array.from(value);
  while (cookie.length) {
    chunks.push({
      name: `${key}.${chunks.length}`,
      value: cookie.splice(0, MAX_CHUNK_SIZE)
    });
  }
```

Reply:
Wouldn't this be a bit slower since it would convert all characters of the value into a string?

Reply:
It'd definitely not be as slow as the regexp, but it's just a matter of taste anyway IMO. It's not a very hot code path.